src/Events/Calendar_Embeds/Admin/Page.php

```diff
@@ -0,0 +1,169 @@
+		$this->hook_suffix = add_submenu_page(
+			'edit.php?post_type=' . TEC::POSTTYPE,
+			esc_html( $this->get_page_title() ),
+			esc_html( $this->get_menu_label() ),
+			$cpt->cap->publish_posts,
+			'edit.php?post_type=' . Calendar_Embeds::POSTTYPE,
+		);
+
+		return $this->hook_suffix;
+	}
+
+	/**
+	 * Gets the hook suffix for the Calendar Embeds.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 * @throws \RuntimeException If the hook suffix is not set.
+	 */
+	public function get_hook_suffix() {
+		if ( ! $this->hook_suffix ) {
+			throw new \RuntimeException( __( 'Attempted to get hook suffix before it was set.', 'the-events-calendar' ) );
+		}
+
+		return $this->hook_suffix;
+	}
+
+	/**
+	 * Gets the URL for the Calendar Embeds.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_url() {
+		return admin_url( 'edit.php?post_type=' . Calendar_Embeds::POSTTYPE );
+	}
+
+	/**
+	 * Gets the Menu label for the Calendar Embeds.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_menu_label() {
+		return __( 'Embed Calendar', 'the-events-calendar' );
+	}
+
+	/**
+	 * Gets the Page title for the Calendar Embeds.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_page_title() {
+		return __( 'Embed Calendar', 'the-events-calendar' );
+	}
+
+	/**
+	 * Keep parent menu open when adding and editing calendar embeds.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $submenu_file The current submenu file.
+	 *
+	 * @return string
+	 */
+	public function keep_parent_menu_open( $submenu_file ) {
+		global $parent_file;
+
+		if ( 'edit.php?post_type=' . Calendar_Embeds::POSTTYPE !== $parent_file ) {
+			return $submenu_file;
+		}
+
+		// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+		$parent_file = 'edit.php?post_type=' . TEC::POSTTYPE;
+
+		return $submenu_file;
+	}
+
+	/**
+	 * Check if the current screen is the Calendar Embeds page.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public static function is_on_page(): bool {
+		$screen = get_current_screen();
+		if ( ! $screen ) {
+			return false;
+		}
+
+		return 'edit-' . Calendar_Embeds::POSTTYPE === $screen->id;
+	}
+
+	/**
+	 * Register assets for the Calendar Embeds page.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function register_assets(): void {
+		Config::set_path( TEC::instance()->plugin_path );
+		Asset::add(
+			'tec-events-calendar-embeds-script',
+			'js/admin/calendar-embeds-page.js'
+		)
+			->enqueue_on( 'admin_enqueue_scripts' )
+			->set_condition( [ __CLASS__, 'is_on_page' ] )
+			->set_dependencies( 'thickbox', 'tribe-clipboard' )
+			->in_footer()
+			->register();
+
+		Asset::add(
+			'tec-events-calendar-embeds-style',
+			'css/admin/calendar-embeds-page.css'
+		)
+			->enqueue_on( 'admin_enqueue_scripts' )
+			->set_condition( [ __CLASS__, 'is_on_page' ] )
+			->set_dependencies( 'thickbox', 'tribe-common-admin' )
+			->register();
+	}
```

Comment:
Set their enqueue_on on the action is being fired by the page's render. So you dont have to worry about the conditional enqueue

Reply:
In this case, we're using the built-in post type page, not rendering it ourselves. That's why I did it this way.

Reply:
Thats good! Thank you!